assets/js/modules/analytics/datastore/settings.js

```diff
@@ -299,3 +299,56 @@ export const isGA4DashboardView = createRegistrySelector( ( select ) => () => {
 
 	return ga4Settings.dashboardView === 'google-analytics-4';
 } );
+
+/**
+ * Determines whether the user should be prompted to switch to GA4 Dashboard View.
+ *
+ * @since n.e.x.t
+ *
+ * @return {boolean} True if the user should be prompted to switch to GA4 Dashboard, false otherwise, or undefined if not loaded.
+ */
+export const shouldPromptGA4DashboardView = createRegistrySelector(
+	( select ) => () => {
+		const isGA4Enabled = isFeatureEnabled( 'ga4Reporting' );
+		if ( ! isGA4Enabled ) {
+			return false;
+		}
+
+		const ga4ModuleConnected =
+			select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+
+		if ( ga4ModuleConnected === undefined ) {
+			return undefined;
+		}
+
+		if ( ! ga4ModuleConnected ) {
+			return false;
+		}
+
+		const ga4DashboardView =
+			select( MODULES_ANALYTICS ).isGA4DashboardView();
+
+		if ( ga4DashboardView === undefined ) {
+			return undefined;
+		}
+
+		// Don't prompt if the user is already on the GA4 Dashboard.
+		if ( ga4DashboardView ) {
+			return false;
+		}
+
+		const ga4GatheringData =
```

Comment:
And once again:
```suggestion
		const isGA4GatheringData =
```

Reply:
See [here](https://github.com/google/site-kit-wp/pull/6752#discussion_r1144902050).